src/agent/configuration_parser/src/configuration_parser.cpp

```diff
@@ -121,4 +106,194 @@ namespace configuration
+                            {
+                                mergedSequence.push_back(overrideItem->second);
+                                mergedKeys.insert(overrideItem->first);
+                            }
+                            else
+                            {
+                                mergedSequence.push_back(elem);
+                            }
+                        }
+
+                        // Add remaining elements from 'override' sequence in order.
+                        for (const auto& [itemKey, itemNode] : overrideElements)
+                        {
+                            if (mergedKeys.find(itemKey) == mergedKeys.end())
+                            {
+                                mergedSequence.push_back(itemNode);
+                            }
+                        }
+
+                        baseNode[key] = mergedSequence;
+                    }
+                    else
+                    {
+                        // Other cases (scalar, alias, null): overwrite the value.
+                        baseNode[key] = value;
+                    }
+                }
+                else
+                {
+                    // Key does not exist in the base node: add it directly.
+                    baseNode[key] = value;
+                }
+            }
+        }
+    }
+
+    void ConfigurationParser::LoadSharedConfig()
+    {
+        LogDebug("Loading shared configuration.");
+        if (m_getGroups == nullptr)
+        {
+            LogWarn("Load shared configuration failed, no get groups function set");
+            return;
+        }
+
+        try
+        {
+            const std::vector<std::string> groupIds = m_getGroups();
+            YAML::Node tmpConfig = m_config;
+
+            for (const auto& groupId : groupIds)
+            {
+                const std::filesystem::path groupFile = std::filesystem::path(config::DEFAULT_SHARED_CONFIG_PATH) /
+                                                        (groupId + config::DEFAULT_SHARED_FILE_EXTENSION);
+
+                LogDebug("Loading group configuration file: {}.", groupFile.string());
+
+                YAML::Node fileToAppend = YAML::LoadFile(groupFile.string());
+
+                if (!tmpConfig.IsDefined() || tmpConfig.IsNull())
+                {
+                    tmpConfig = fileToAppend;
+                }
+                else
+                {
+                    MergeYamlNodes(tmpConfig, fileToAppend);
+                }
+            }
+
+            m_config = tmpConfig;
+        }
+        catch (const YAML::Exception& e)
+        {
+            LogWarn("Load shared configuration failed: {}", e.what());
+        }
+    }
+
+    void ConfigurationParser::SetGetGroupIdsFunction(std::function<std::vector<std::string>()> getGroupIdsFunction)
+    {
+        m_getGroups = std::move(getGroupIdsFunction);
+        LoadSharedConfig();
```

Comment:
The function should just be a setter for the function, its used should be separate, do we have use cases where we call this function and we can't call `LoadSharedConfig`?

Reply:
Done!